pkg/compute/serve.go

```diff
@@ -0,0 +1,287 @@
+		return filepath.Join(dir, "fastly")
+	}
+	if dir, err := os.UserHomeDir(); err == nil {
+		return filepath.Join(dir, ".fastly")
+	}
+	panic("unable to deduce user config dir or user home dir")
+}()
+
+// installViceroy downloads the latest release from GitHub.
+func installViceroy(progress text.Progress, versioner update.Versioner, latest semver.Version, bin string) error {
+	progress.Step("Fetching latest viceroy release...")
+
+	tmp, err := versioner.Download(context.Background(), latest)
+	if err != nil {
+		progress.Fail()
+		return fmt.Errorf("error downloading latest viceroy release: %w", err)
+	}
+
+	if err := os.Rename(tmp, bin); err != nil {
+		if err := filesystem.CopyFile(tmp, bin); err != nil {
+			progress.Fail()
+			return fmt.Errorf("error moving latest viceroy binary in place: %w", err)
+		}
+	}
+
+	return nil
+}
+
+// updateViceroy checks if the currently installed version is out-of-date and
+// downloads the latest release from GitHub.
+func updateViceroy(progress text.Progress, version string, out io.Writer, versioner update.Versioner, latest semver.Version, bin string) error {
+	progress.Step("Checking installed viceroy version...")
+
+	var installedViceroyVersion string
+
+	viceroyError := errors.RemediationError{
+		Inner:       fmt.Errorf("viceroy version not found"),
+		Remediation: errors.BugRemediation,
+	}
+
+	// version output has the expected format: `viceroy 0.1.0`
+	segs := strings.Split(version, " ")
+
+	if len(segs) < 2 {
+		return viceroyError
+	}
+
+	installedViceroyVersion = segs[1]
+
+	if installedViceroyVersion == "" {
+		return viceroyError
+	}
+
+	current, err := semver.Parse(installedViceroyVersion)
+	if err != nil {
+		progress.Fail()
+
+		return errors.RemediationError{
+			Inner:       fmt.Errorf("error reading current version: %w", err),
+			Remediation: errors.BugRemediation,
+		}
+	}
+
+	if latest.GT(current) {
+		text.Output(out, "Current viceroy version: %s", current)
+		text.Output(out, "Latest viceroy version: %s", latest)
+
+		progress.Step("Fetching latest viceroy release...")
+		tmp, err := versioner.Download(context.Background(), latest)
+		if err != nil {
+			progress.Fail()
+			return fmt.Errorf("error downloading latest viceroy release: %w", err)
+		}
+		defer os.RemoveAll(tmp)
+
+		progress.Step("Replacing viceroy binary...")
+
+		if err := os.Rename(tmp, bin); err != nil {
+			if err := filesystem.CopyFile(tmp, bin); err != nil {
+				progress.Fail()
+				return fmt.Errorf("error moving latest viceroy binary in place: %w", err)
+			}
+		}
+	}
+
+	return nil
+}
+
+// local spawns a subprocess that runs the compiled binary.
+func local(bin string, out io.Writer, env string, verbose bool) error {
+	sig := make(chan os.Signal, 1)
+
+	signals := []os.Signal{
+		syscall.SIGINT,
+		syscall.SIGTERM,
+	}
+
+	signal.Notify(sig, signals...)
+
+	if env != "" {
+		env = "." + env
+	}
+	manifest := fmt.Sprintf("fastly%s.toml", env)
+	args := []string{"bin/main.wasm", "-C", manifest}
```

Comment:
Not an issue now, but something for us to consider, we're making a lot of assumptions here that `fastly compute serve` was run in the root directory of a project. Is this ok? Do we want users to provide a path to teh manifest and binary via flags so it can be ran elsewhere, this may be useful in CI scenarios.